File: apps/clinical_diary/functions/src/health.ts
/**
 * IMPLEMENTS REQUIREMENTS:
 *   REQ-d00005: Sponsor Configuration Detection Implementation
 *   REQ-p00013: GDPR compliance - EU-only regions
 *
 * Health check endpoint for Firebase Cloud Functions.
 */

import * as functions from "firebase-functions/v1";
import {corsHandlerFnc} from "./cors";
import {runtimeOpts} from "./index";

/**
 * Health check HTTP function (v1)
 */
export const health = functions
  .runWith(runtimeOpts)
  .region("europe-west1")
  .https.onRequest((req, res) => {
    corsHandlerFnc()(req, res, async () => {
      res.json({
        status: "ok",
        timestamp: new Date().toISOString(),
        region: "europe-west1",
      });
    });
  });

File: apps/daily-diary/diary_functions/lib/src/health.dart
// IMPLEMENTS REQUIREMENTS:
//   REQ-d00005: Sponsor Configuration Detection Implementation
//   REQ-p00013: GDPR compliance - EU-only regions
//
// Health check handler - converted from Firebase health.ts

import 'dart:convert';

import 'package:shelf/shelf.dart';

/// Health check endpoint handler
/// Returns server status for Cloud Run health checks
Response healthHandler(Request request) {
  final body = jsonEncode({
    'status': 'ok',
    'timestamp': DateTime.now().toUtc().toIso8601String(),
    'region': 'europe-west1',
    'service': 'diary-server',
  });

  return Response.ok(body, headers: {'Content-Type': 'application/json'});
}
